### PR TITLE
Responsiveapi

### PR DESCRIPTION
**Pull request by Awzaw**
I think this is ready. Not perfect... but ready.

### CODE REVIEW

File: entry.php [hunk 1]
```diff
@@ -150,8 +151,9 @@ function error_handler(int $errno, string $error, string $errfile, int $errline)
         http_response_code(500);
         $refid = mt_rand();
         if(Poggit::$plainTextOutput) {
+            header("Content-Type: text/plain");
             OutputManager::$current->outputTree();
-            echo "Error#$refid Level $errno error at $errfile:$errline: $error\n";
+            echo "Error#$refid" /* " Level $errno error at $errfile:$errline" . */ . ": $error\n";
```
Comment (SOF3) [inline entry.php]: :+1:

Delete the line directly too.

File: js/std.js [hunk 2]
```diff
@@ -30,6 +30,16 @@ String.prototype.ucfirst = function() {
     return this.charAt(0).toUpperCase() + this.substr(1)
 };
 
+/**
+ * No OPeration placeholder function
+ */
+function nop() {
+}
+
+function isLoggedIn() {
```
Comment (SOF3) [inline js/std.js]: This is already on line 279

File: src/poggit/debug/AddResourceModule.php [hunk 3]
```diff
@@ -28,6 +28,7 @@ public function getName(): string {
     }
 
     public function output() {
+        parent::output();
```
Comment (SOF3) [inline src/poggit/debug/AddResourceModule.php]: :+1:

File: src/poggit/module/build/RecentBuildPage.php [hunk 4]
```diff
@@ -37,12 +37,12 @@ public function getTitle(): string {
     }
 
     public function output() {
-        if($this->error !== "") {
+        if ($this->error !== "") {
```
Comment (SOF3) [inline src/poggit/module/build/RecentBuildPage.php]: hmm?

File: src/poggit/module/build/RepoListBuildPage.php [hunk 5]
```diff
@@ -104,9 +104,9 @@ protected function displayRepos(array $repos) {
             $opened = "false";
             if(count($repo->projects) === 1) $opened = "true";
             ?>
-            <div class="toggle" data-name="<?= $repo->full_name ?> (<?= count($repo->projects) ?>)"
+            <div class="repotoggle" data-name="<?= $repo->full_name ?> (<?= count($repo->projects) ?>)"
                  data-opened="<?= $opened ?>">
-                <h2><a href="<?= $home ?>ci/<?= $repo->full_name ?>">Projects</a> in
+                <h2><a href="<?= $home ?>ci/<?= $repo->full_name ?>"></a>
```
Comment (SOF3) [inline src/poggit/module/build/RepoListBuildPage.php]: So now what does this hyperlink do?

File: src/poggit/module/build/SelfBuildPage.php [hunk 6]
```diff
@@ -1,5 +1,4 @@
 <?php
-
```
Comment (SOF3) [inline src/poggit/module/build/SelfBuildPage.php]: what?

File: src/poggit/module/build/RecentBuildPage.php (hunk 4 above)
Comment (Awzaw) [inline src/poggit/module/build/RecentBuildPage.php]: not deliberate...

File: src/poggit/module/build/SelfBuildPage.php (hunk 6 above)
Comment (Awzaw) [inline src/poggit/module/build/SelfBuildPage.php]: If you look at a fully populated page it's so obvious that they are projects there's no need to repeat the word 20 times as, not as a heading at least, or at the start of the box

Comment (Awzaw) [inline src/poggit/module/build/SelfBuildPage.php]: no idea...

File: src/poggit/module/build/RepoListBuildPage.php (hunk 5 above)
Comment (Awzaw) [inline src/poggit/module/build/RepoListBuildPage.php]: If you look at a fully populated page it's so obvious that they are projects there's no need to repeat the word 20 times, not as a heading at least, or as the first word. the link is maybe useful which might be why I didn't remove it completely... but the text itself offers no information. The same goes for "Repo Enabled? Change", it's not necessary and just clutters things up

File: entry.php (hunk 1 above)
Comment (Awzaw) [inline entry.php]: ok

File: src/poggit/module/build/RepoListBuildPage.php (hunk 5 above)
Comment (SOF3) [inline src/poggit/module/build/RepoListBuildPage.php]: I suppose we can still add a link to the RepoBuildPage for that repo? Just like the GitHub logo link.

**Comment by SOF3** [comment]
Merged ef4a2568282ae5fa8a4d620daf9b14f3e20fbc7b

**Pull request closed without merging**